### ISSUE TITLE
[tune] reuse_actors doesn't reuse actors for the class based API

### ISSUE THREAD

**Issue by cruyffturn** [ray-project/ray#13475]
### What is the problem?
"reuse_actors = True" doesn't work as expected for the class based API. When I inspect the timeline I observe that workers are not reused. On the other hand, when I use the function based API it works faster and only 3 workers are used in the timeline as expected.

Ray version 1.1.0, Python 3.6.12.

### Reproduction (REQUIRED)
```
import time
import ray
from ray import tune
import numpy as np

class Trainable(tune.Trainable):
    
    def _setup(self, config):
        self.i = 0
    
    def _train(self):  # This is called iteratively.
        print(self.i)
        time.sleep(1)
        self.i += 1
            
        return {'i':self.i}    

ray.init(num_cpus=3, num_gpus=0)
        
search_space = {
        "i": tune.grid_search(np.arange(20).tolist())
        }

analysis = tune.run(Trainable,
                    name = 'exp',
                    stop = {'i':1},
                    config=search_space,
                    reuse_actors = True
                    )
```

- [y] I have verified my script runs in a clean environment and reproduces the issue.
- [y] I have verified the issue also occurs with the [latest wheels](https://docs.ray.io/en/master/installation.html).

**Comment by krfricke** [comment]
In order to use `reuse_actors` with the class based API you need to implement a `reset_config()` method - see here: https://docs.ray.io/en/master/tune/api_docs/trainable.html#advanced-reusing-actors

Your output should also show a warning that this is not the case.

Please feel free to re-open if you have more questions.

**Comment by cruyffturn** [comment]
Thank you. I apologize, I missed that.

**Comment by cruyffturn** [comment]
@krfricke I have included the reset_config method that returns true but from the timeline I observe the actors are not reused.

What can be the problem?

### Reproduction (REQUIRED)
```
import time
import ray
from ray import tune
import numpy as np

class Trainable(tune.Trainable):
    
    def setup(self, config):
        self.i = 0
    
    def step(self):
        print(self.i)
        time.sleep(1)
        self.i += 1
            
        return {'i':self.i}
    
    def reset_config(self, new_config):
        
        self.config = new_config
        return True

ray.init(num_cpus=3, num_gpus=0)
        
search_space = {
        "i": tune.grid_search(np.arange(20).tolist())
        }

analysis = tune.run(Trainable,
                    name = 'exp',
                    stop = {'i':1},
                    config=search_space,
                    reuse_actors = True
                    )
```

**Comment by krfricke** [comment]
Hi @cruyffturn, there seems to be the requirement that the trial has to start from a checkpoint in order to reuse existing trainables. This might be a legacy requirement that we can get rid of. I'm actively looking into this for a similar case, so I'll keep you posted with our progress there.

**Comment by cruyffturn** [comment]
Thank you!

**Comment by krfricke** [comment]
Hi @cruyffturn, this was fixed in #13549 and will be included in the next nightly wheels and next release.
